assets/src/edit-story/elements/text/updateForResizeEvent.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { MIN_FONT_SIZE, MAX_FONT_SIZE } from '../../constants';
+import {
+  calculateFitTextFontSize,
+  calculateTextHeight,
+} from '../../utils/textMeasurements';
+import { dataPixels } from '../../units/dimensions';
+
+function updateForResizeEvent(element, newWidth, newHeight) {
```

Comment:
Q: Will any other element have this, too? Wondering if it makes sense as a type definition or as an util. Thoughts?

Reply:
It seemed like type definition was likely. But I initially built it this way to allow a deeper understanding of display DOM tree. Later, however, I changed to have a separate hidden DOM element to be used for all calculations. Now, this callback looks more like a reducer. I think we should keep it in type declaration and see what happens to shapes - the other element that will likely need type-based resize rules. Then we will have a better view on where it fits better.